projects/core/styles/components/textfield.less

```diff
@@ -102,7 +102,8 @@ tui-textfield {
         }
     }
 
-    &:hover {
+    &:hover:has(input:not(:read-only)),
```

Comment:
Let's invert this to the other way around. With this approach non-readonly inputs icons will not change colors in old Safari. If instead we would enforce static color for such icons, it would just be a graceful degradation — icons will still change colors when they shouldn't in old Safari, it is better than not changing colors when they should.

Reply:
@waterplea Do you mean the following approach ?

```less
tui-textfield:hover {
    color: var(--tui-text-secondary);
    
    &:has(input:not(:read-only)),
    &:has(select:not([data-mode='readonly'])) {
        color: var(--tui-text-tertiary);
    }
}
```

Reply:
fixed in https://github.com/taiga-family/taiga-ui/pull/8754/commits/933ff600c39954725a06e4cddf04518c5cf5e3e8